test/e2e/usersignup_test.go

```diff
@@ -197,7 +197,7 @@ func (s *userSignupIntegrationTest) TestProvisionToOtherClusterWhenOneIsFull() {
 			Execute(s.T()).Resources()
 
 		// then
-		require.NotEqual(t, mur1.Spec.UserAccounts[0].TargetCluster, mur2.Spec.UserAccounts[0].TargetCluster)
+		require.NotEqual(t, mur1.Status.UserAccounts[0].Cluster.Name, mur2.Status.UserAccounts[0].Cluster.Name)
```

Comment:
I couldn't find where `mur.Status.UserAccounts.Cluster.Name` is being set in the host-operator and member-operator, I see it's being checked but not set. I'm sure I've missed something. 

But should we use `hostAwait.WaitForSpace(t, mur1.Name, wait.UntilSpaceHasAnyTargetClusterSet())` and same for `mur2` here as well ?

Reply:
> I couldn't find where mur.Status.UserAccounts.Cluster.Name is being set in the host-operator and member-operator, I see it's being checked but not set. I'm sure I've missed something.

I'm not sure if I follow what you mean. Don't forget that this is a status, so it's set by a MUR controller when it syncs the information with UserAccount https://github.com/codeready-toolchain/host-operator/blob/927edad1f670889d02485c1536a19009bec48e18/controllers/masteruserrecord/sync.go#L95

> But should we use hostAwait.WaitForSpace(t, mur1.Name, wait.UntilSpaceHasAnyTargetClusterSet()) and same for mur2 here as well ?

I probably don't follow your message again. Do you mean that we should have an extra check that would wait until the Space has the needed targetCluster set, etc...? This is already done in the "SignupRequest" fluent API logic - we call it with `EnsureMUR()` which ensures both the MUR and Space are fully provisoned. I know that the name is misleading - it's another thing to improve but in separate PR, I don't want to mix it with the functional changes.

Reply:
> I'm not sure if I follow what you mean. Don't forget that this is a status, so it's set by a MUR controller when it syncs the information with UserAccount https://github.com/codeready-toolchain/host-operator/blob/927edad1f670889d02485c1536a19009bec48e18/controllers/masteruserrecord/sync.go#L95

Ok I see now that the `mur.Status.UserAccount.Cluster.Name` is being set for the fist time here: I'm not sure if I follow what you mean. Don't forget that this is a status, so it's set by a MUR controller when it syncs the information with UserAccount https://github.com/codeready-toolchain/host-operator/blob/927edad1f670889d02485c1536a19009bec48e18/controllers/masteruserrecord/sync.go#L95

And the UserAccount is being added to the status or updated based on `index` < `0` here:https://github.com/codeready-toolchain/host-operator/blob/927edad1f670889d02485c1536a19009bec48e18/controllers/masteruserrecord/sync.go#L103-L107

Thanks for the pointer!

Reply:
> I probably don't follow your message again. Do you mean that we should have an extra check that would wait until the Space has the needed targetCluster set, etc...? 

Basically I was wondering if we should rely on this `MUR.Status.UserAccounts[0].Cluster.Name` field or if we should get the Spaces for both mur1 and mur1 like we do here for example https://github.com/codeready-toolchain/toolchain-e2e/pull/807/files#diff-d5d12b6ea07b0a78c9ad1987e1486c565ecf16839f77a93e91b09475cf6c0e99R148 and use the `Space.Spec.TargetCluster` for the comparison. But I guess this `MUR.Status.UserAccounts[0].Cluster.Name` will always be up-to-date.

Reply:
yeah, good idea: done here d424d5ae

Reply:
Nice! Makes sense to have it in addition to the existing check 👍 

Thanks!